src/tools/pub/automated-publishing.md

```diff
@@ -0,0 +1,615 @@
+[default Cloud Build service account][13] permission to impersonate 
+the service account created for publishing in the previous section.
+
+1. Enable the [IAM Service Account Credentials API][14] in the cloud project.
+   Attempts to impersonate a service account will fail without this API.
+
+   ```sh
+   # Enable IAM Service Account Credentials API
+   gcloud services enable iamcredentials.googleapis.com
+   ```
+
+1. Find the project number.
+
+   ```sh
+   # The PROJECT_NUMBER can be obtained as follows:
+   gcloud projects describe $PROJECT_ID --format='value(projectNumber)'
+   ```
+
+1. Grant the permission to impersonate the publishing service account.
+
+   ```sh
+   # Grant default cloud
+   gcloud iam service-accounts add-iam-policy-binding \
+     pub-dev@$PROJECT_ID.iam.gserviceaccount.com \
+     --member=serviceAccount:$PROJECT_NUMBER@cloudbuild.gserviceaccount.com \
+     --role=roles/iam.serviceAccountTokenCreator
+   ```
+
+### Writing a Cloud Build Configuration File
+
+To publish from Cloud Build, we must specify steps for Cloud Build to:
+ * Impersonate the service account to obtain a temporary OIDC token.
+ * Provide the temporary OIDC token to `dart pub` for use when publishing
+ * Calling `dart pub publish` to publish the package
+
+Steps for Google Cloud Build are provided in a `cloudbuild.yaml` file, see
+[build configuration file schema][15] for full documentation of the format.
+
+For publishing to pub.dev from Google Cloud Build, a `cloudbuild.yaml` file as
+follows will do:
+
+```yaml
+# cloudbuild.yaml
+steps:
+- id: Create temporary token
+  name: gcr.io/cloud-builders/gcloud
+  volumes:
+  - name: temporary-secrets
+    path: /secrets
+  script: |
+    gcloud auth print-identity-token \
+      --impersonate-service-account=pub-dev@$PROJECT_ID.iam.gserviceaccount.com \
+      --audiences=https://pub.dev \
+      --include-email > /secrets/temporary-pub-token.txt
+  env:
+  - PROJECT_ID=$PROJECT_ID
+- id: Publish to pub.dev
+  name: dart
+  volumes:
+  - name: temporary-secrets
+    path: /secrets
+  script: | 
+    cat /secrets/temporary-pub-token.txt | dart pub token add https://pub.dev
+    dart pub publish --force
+```
+
+The `gcloud auth print-identity-token` creates an OIDC `id_token` impersonating
+the specified service account. This `id_token` is signed by Google, with a
+signature that expires within 1 hour. The _audiences_ parameters lets pub.dev
+know that it is the intended recipient of the token. The `--include-email`
+option is necessary for pub.dev to be able to recognize the service account.
+
+Once the `id_token` is created it is written to a file that resides in a
+_volume_, this mechanism is used to [pass data between steps][16].
+Don't store the token in `/workspace`. Since `/workspace` is where the
+repository from which you wish to publish is checked out.
+Not using `/workspace` for storing the token, reduces the risk that you
+accidentally include it in your package when publishing.
```

Comment:
```suggestion
Once the `id_token` is created, it's written to a file that resides in a
_volume_; this mechanism is used to [pass data between steps][16].
Don't store the token in `/workspace`. Since `/workspace` is where the
repository from which you wish to publish is checked out.
Not using `/workspace` for storing the token reduces the risk that you
accidentally include it in your package when publishing.
```